src/transformers/modeling_tf_auto.py

```diff
@@ -896,6 +938,409 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         )
 
 
+class TFAutoModelForCausalLM:
```

Comment:
Analog to PyTorch we deprecate `TFAutoModelWithLMHead` here and add `TFAutoModelForCausalLM` `TFAutoModelForSeq2Seq`, `TFAutoModelForMaskedLM` .